errors_test.go

```diff
@@ -112,7 +121,46 @@ func TestWrap(t *testing.T) {
 	assert.Equal(t, wrappedErr.Params, map[string]string{
 		"blub": "dub",
 	})
+}
+
+func TestWrapErrorWithMessage(t *testing.T) {
+	err := fmt.Errorf("Look here, an error")
+	wrappedErr := Wrap(err, nil, "loading data").(*Error)
+
+	assert.Equal(t, "internal_service: loading data: Look here, an error", wrappedErr.Error())
+	assert.Equal(t, "loading data: Look here, an error", wrappedErr.Message)
+}
+
+func TestWrapTerrorWithMessage(t *testing.T) {
+	err := BadRequest("name", "invalid name", nil)
+	wrappedErr := Wrap(err, nil, "loading data").(*Error)
+
+	assert.Equal(t, "bad_request.name: loading data: invalid name", wrappedErr.Error())
+	assert.Equal(t, "loading data: invalid name", wrappedErr.Message)
+}
+
+func TestWrappedErrorCanBeUnwrapped(t *testing.T) {
+	err := &customError{"foo"}
+	wrappedErr := Wrap(err, nil)
+
+	assert.True(t, errors.Is(wrappedErr, err))
```

Comment:
Using `errors.Is` and `errors.As` in these test files mean that we can't support versions of Go < 1.13. If we wanted to maintain backwards compatibility we could instead test the `Unwrap` method directly without reference to `errors.Is` and `errors.As`.

Reply:
I think we're fine to assume >= 1.13